server/protocols/protocols-smtp/src/main/java/org/apache/james/smtpserver/tls/TestMtaSts.java

```diff
@@ -0,0 +1,50 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.smtpserver.tls;
+
+import org.apache.james.core.MaybeSender;
+import org.apache.james.protocols.smtp.SMTPSession;
+import org.apache.james.protocols.smtp.hook.HookResult;
+import org.apache.james.protocols.smtp.hook.MailHook;
+import org.apache.james.util.MDCStructuredLogger;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * CF https://datatracker.ietf.org/doc/html/rfc8461
+ *
+ * SMTP MTA Strict Transport Security (MTA-STS)
+ *
+ * Aimed at enforcing mode testing
+ */
+public class TestMtaSts implements MailHook {
```

Comment:
I don't understand what this class does. it logs a warning if TLS is not started and does not return an error code but it returns a DECLINED

the implications of each HookResultCode actions is not very well documented and the code is not the most explicit. I think this will ignore the command without returning an error. 

Is this really the behaviour we want when testing the application of MTA STS ? I would expect the hook to warn but accept the command instead. 
this would let the administrator investigate how much traffic does not honor the published MTA STS policy without disrupting the traffic. 

what did it miss ?

Reply:
The idea was to log when TLS is not active upon init of a transaction.

We want to warn the admin of non compliant connections, so that the admin can understand the impact of enforcing MTA-STS on his upcoming traffic.

I'll update the doc.

Reply:
DECLINED means "skip me" in SMTP hooks.

Terrible syntax, I agree, I always get confused TBH...

But I won't fix this here...